lib/post_destroyer.rb

```diff
@@ -43,7 +43,14 @@ def self.delete_with_replies(performed_by, post, reviewable = nil, defer_reply_f
     reply_ids = post.reply_ids(Guardian.new(performed_by), only_replies_to_single_post: false)
     replies = Post.where(id: reply_ids.map { |r| r[:id] })
     PostDestroyer.new(performed_by, post, reviewable: reviewable).destroy
-    replies.each { |reply| PostDestroyer.new(performed_by, reply, defer_flags: defer_reply_flags).destroy }
+
+    options = { defer_flags: defer_reply_flags }
+    if SiteSetting.notify_users_after_responses_deleted_on_flagged_post
+      options[:reviewable] = reviewable
+      options[:notify_responders] = true
+      options[:parent_post] = post
+    end
```

Comment:
I wanted to short hand this with `.merge` and had no luck. I was not able to identify why that would not work. eg

```
options.merge({ reviewable: reviewable, notify_responders: true, parent_post: post }) if SiteSetting.notify_users_after_responses_deleted_on_flagged_post
```

Reply:
`#merge` returns a new hash so you need to re-assign the variable, or call `#merge!` instead, which mutates the original hash.